src/structures/paging/recursive.rs

```diff
@@ -1,21 +1,51 @@
 use registers::control::Cr3;
+use instructions::tlb;
 use structures::paging::page_table::{FrameError, PageTable, PageTableEntry, PageTableFlags};
 use structures::paging::{NotGiantPageSize, Page, PageSize, PhysFrame, Size1GB, Size2MB, Size4KB};
 use ux::u9;
 use VirtAddr;
 
+use core::mem;
+
+/// This type must be used and will either flush the modified page or can be unsafely ignored.
+#[must_use = "Page Table changes must be flushed or unsafely ignored."]
+pub struct MapperFlush<S: PageSize>(Page<S>);
+
+impl<S: PageSize> MapperFlush<S> {
+    /// Create a new flush promise
+    fn new(page: Page<S>) -> Self {
+        MapperFlush(page)
+    }
+
+    // Flush 
+    pub fn flush(self) {
+        tlb::flush(self.0.start_address());
+        mem::forget(self);
```

Comment:
Why do we need mem::forget here?

Reply:
This was so that drop would not get called.

Reply:
Why? `Page` is a `Copy` type without a destructor, so nothing should happen when it's dropped.

Reply:
But `MapperFlush` did have a destructor that would always panic, so that needed to be avoided.

Reply:
Oh, makes sense :)

Reply:
Easy mistake to make :)